cli/src/config.rs

```diff
@@ -17,9 +17,18 @@ To learn more, checkout https://apollo.dev/cli/telemetry\n";
 #[derive(Debug, Serialize, Deserialize)]
 pub struct CliConfig {
     pub machine_id: String,
+    pub api_key: Option<String>,
 }
 
 impl CliConfig {
+    pub fn write(cli_config: &CliConfig) -> Result<(), Box<dyn Error + 'static>> {
```

Comment:
I think we could split this into two functions:
`pub fn save() -> Result`
`pub fn create_config_file()` -> Result`

Save should check for the file to exist before any write and, if needed, create the dir structure and file.